### PR TITLE
add Dockerfiles for each app

### PR DESCRIPTION
**Pull request by rgitzel**

### CODE REVIEW

**Comment by robertreppel** [comment]
https://www.reddit.com/r/docker/comments/73zxle/why_files_created_from_my_container_owned_by_root/ 

... this looks related.

**Comment by robertreppel** [comment]
Tried it on another Ubuntu 16.04 machine. Same result. The container writes to the mapped volume as "root."

**Comment by rgitzel** [comment]
@robertreppel I did some digging, seems the way to do this is to explicitly provide a user in the Dockerfile.  Fair enough, but the catch is it has to match the user and group id of the user that runs the Docker image. Which seems odd.

I'll do some more digging, but in the meantime if you want to try the latest version, and you change the two IDs in the Dockerfile, it should no longer be writing files as roo.

File: samples/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.generated.eml.yaml
```
Comment (rgitzel) [inline samples/.gitignore:1]: I noticed this being left behind, not sure if it should be cleaned up by the app.

**Comment by rgitzel** [comment]
@robertreppel It should be ready to go. It defaults to USER ID 1000, which seems pretty typical, and can be overridden (as noted in the README).

**Comment by robertreppel** [comment]
@rgitzel sounds good ... I'll give it a go!

File: samples/.gitignore (hunk 1 above)
Comment (robertreppel) [reply samples/.gitignore:1]: So far I've been leaving them behind deliberately because they are handy for exploring the generated EML.

Comment (robertreppel) [reply samples/.gitignore:1]: Could just as well do "esp convert" to see that, though.

**Comment by robertreppel** [comment]
Merged it. Thanks.

**Pull request closed without merging**